pkg/apis/extensions/validation/validation.go

```diff
@@ -293,7 +293,10 @@ func ValidateDeploymentSpec(spec *extensions.DeploymentSpec) errs.ValidationErro
 	allErrs = append(allErrs, apivalidation.ValidatePositiveField(int64(spec.Replicas), "replicas")...)
 	allErrs = append(allErrs, apivalidation.ValidatePodTemplateSpecForRC(&spec.Template, spec.Selector, spec.Replicas, "template")...)
 	allErrs = append(allErrs, ValidateDeploymentStrategy(&spec.Strategy, "strategy")...)
-	allErrs = append(allErrs, apivalidation.ValidateLabelName(spec.UniqueLabelKey, "uniqueLabel")...)
+	// empty string is a valid UniqueLabelKey
```

Comment:
API allows users to set it to empty string to indicate that no label should be added. The default value will be set if user does not set it, not if user sets it to empty string (its a string pointer)
From API comments:
>
        // Users can set this to an empty string to indicate that the system should
	// not add any selector and label. If unspecified, system uses
	// "deployment.kubernetes.io/podTemplateHash".
	// Value of this key is hash of DeploymentSpec.PodTemplateSpec.
	// No label is added if this is set to empty string.
>